packages/playwright-core/src/server/webkit/wkPage.ts

```diff
@@ -571,6 +571,16 @@ export class WKPage implements PageDelegate {
         context = this._contextIdToContext.get(objectId.injectedScriptId);
       } else {
         context = this._contextIdToContext.get(this._mainFrameContextId!);
+        if (!this._page._browserContext._options.javaScriptEnabled) {
```

Comment:
I think we just want to grab any context in the frame, not necessary main context or when javascript is disabled.

```ts
} else {
  context = [...this._contextIdToContext.values()].find(c => c.frame === this._page.mainFrame());
}
```